clients/drcachesim/tracer/output.cpp

```diff
@@ -1022,9 +1037,17 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
             }
             if (hit_window_end) {
                 if (op_offline.get_value() && op_split_windows.get_value()) {
-                    size_t add =
-                        instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
-                    buf_ptr += add;
+                    if (op_L0_filter_until_instrs.get_value() &&
+                        mode == BBDUP_MODE_L0_FILTER) {
+                        NOTIFY(0, "Adding endpoint marker\n");
+                        size_t add = instru->append_marker(
```

Comment:
So we're making the switch from L0_filter to full trace only at window ends?

Reply:
Updated code to work with other tracing modes also (e.g., -exit_after_tracing, -max_trace_size/refs)